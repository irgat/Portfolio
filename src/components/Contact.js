import React from 'react';
import PropTypes from 'prop-types';

const errorMessage = `One of the props, 'mailto' or 'tel' is required.`;

const propTypes = {
    mailto: (props, propName, componentName) => {
        if (props.mailto) {
            PropTypes.checkPropTypes({ mailto: PropTypes.string }, { mailto: props.mailto }, propName, componentName);
        } else if (!props.tel) {
            return new Error(errorMessage);
        }
    },
    tel: (props, propName, componentName) => {
        if (props.tel) {
            PropTypes.checkPropTypes({ tel: PropTypes.string }, { tel: props.tel }, propName, componentName);
        } else if (!props.mailto) {
            return new Error(errorMessage);
        }
    },
};

const defaultProps = {

};

const Contact = (props) => {
    let protocol = props.mailto ? 'mailto:' : 'tel:';
    let ref = props.mailto || props.tel;
    return (
        <a href={protocol + ref}>
            <p>{ref}</p>
        </a>
    );
};

Contact.propTypes = propTypes;
Contact.defaultProps = defaultProps;

export default Contact;
