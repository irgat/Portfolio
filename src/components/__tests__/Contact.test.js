import React from 'react';
import Enzyme from 'enzyme';
import Contact from '../Contact';

const data = [
    { mailto: 'local-part@domain' },
    { tel: "XXXXX XXXXXX" },
];

const contact = (props) => {
    return Enzyme.shallow(<Contact {...props} />);
};

describe.each(data)('<Contact />', (props) => {
    it('renders an <a>', () => {
        expect(contact(props).find('a').getElements()).toMatchSnapshot();
    });

    describe('<a />', () => {
        it('renders a <p>', () => {
            expect(contact(props).find('p')).toHaveLength(1);
        });
    });
});
