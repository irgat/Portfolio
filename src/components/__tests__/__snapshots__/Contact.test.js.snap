// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> renders an <a> 1`] = `
Array [
  <a
    href="mailto:local-part@domain"
  >
    <p>
      local-part@domain
    </p>
  </a>,
]
`;

exports[`<Contact /> renders an <a> 2`] = `
Array [
  <a
    href="tel:XXXXX XXXXXX"
  >
    <p>
      XXXXX XXXXXX
    </p>
  </a>,
]
`;
